plugins/aea-ledger-ethereum/aea_ledger_ethereum/ethereum.py

```diff
@@ -518,12 +573,24 @@ def get_transfer_transaction(  # pylint: disable=arguments-differ
             "to": destination_address,
             "value": amount,
             "gas": tx_fee,
-            "gasPrice": self._api.toWei(gas_price, GAS_ID),
+            "gasPrice": gas_price,
             "data": tx_nonce,
         }
         transaction = self.update_with_gas_estimate(transaction)
         return transaction
 
+    @try_decorator("Unable to retrieve gas price: {}", logger_method="warning")
+    def _try_get_gas_price(
+        self, gas_price_strategy: Optional[str] = None
+    ) -> Optional[int]:
+        """Try get the gas price based on the provided strategy."""
+        gas_price_strategy_callable = get_gas_price_strategy(
+            gas_price_strategy, self._gas_price_api_key
+        )
+        self._api.eth.setGasPriceStrategy(gas_price_strategy_callable)
+        gas_price = self._api.eth.generateGasPrice()
```

Comment:
does the previous price strategy get restored? If not, the best would be to save the previous gas price strategy and restore it in the `finally` clause.